src/js/legend.js

```diff
@@ -25,19 +25,61 @@ https://raw.githubusercontent.com/fluid-project/chartAuthoring/master/LICENSE.tx
             // dataSet accepts:
             // 1. an array of objects. Must contain "id", "value" and "label" variables.
             // Example: [{id: string, value: number, label: string} ... ]
-            dataSet: []
+            dataSet: [],
+            total: {
+                // value: number,
+                // percentage: number
+            }
         },
+        modelRelay: [{
```

Comment:
This same ```modelRelay``` block is repeated in three components: dataEntry, pie and legend. It can be done in a most sensible component, probably dataEntry, and use model relay to link up the paths from dataEntry to pieChart, then from pieChart to pie and legend.